File: Visitor/PHP/AddPointVisitor.php
<?php
  class AddPointVisitor implements UserVisitor {
    public function addPointForVip($vip_user) {
      $vip_user->addPoint(10);
      echo "给VIP{$vip_user->getName()}加10分，总分{$vip_user->getPoint()}分\n";
    }

    public function addPointForNormal($normal_user) {
      $normal_user->addPoint(5);
      echo "给普通用户{$normal_user->getName()}加5分，总分{$normal_user->getPoint()}分\n";
    }
  }

File: Visitor/Go/visitor/AddPointVisitor.go
package visitor

import "fmt"

type AddPointVisitor struct {
}

func (this *AddPointVisitor) AddPointForVip(vip_user *VipUser) {
  vip_user.AddPoint(10)
  fmt.Printf("给VIP%s加10分，总分%d分\n", vip_user.GetName(), vip_user.GetPoint())
}

func (this *AddPointVisitor) AddPointForNormal(normal_user *NormalUser) {
  normal_user.AddPoint(5)
  fmt.Printf("给普通用户%s加5分，总分%d分\n", normal_user.GetName(), normal_user.GetPoint())
}
